pkgs/misc/cups/drivers/kyodialog/default.nix

```diff
@@ -0,0 +1,109 @@
+
+  dontStrip = true;
+
+  src = let
+    date = lib.lists.last (builtins.splitVersion version);
+  in fetchzip {
+    url = "https://www.kyoceradocumentsolutions.us/content/download-center-americas/us/drivers/drivers/KyoceraLinuxPackages_${date}_tar_gz.download.gz";
+    sha256 = "sha256-WCHuAQO2T6S8JtRsI2Yf/ypeCLMH365Ax/qX+Ah2s3k=";
+    extension = "tar.gz";
+    stripRoot = false;
+    postFetch = ''
+      # delete redundant Linux package dirs to reduce size in the Nix store; only keep Debian
+      rm -r $out/{CentOS,Fedora,OpenSUSE,Redhat,Ubuntu}
+    '';
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook python3Packages.wrapPython ]
+  ++ lib.optionals withQtGui [ qt5.wrapQtAppsHook ];
+
+  buildInputs = [ cups ] ++ lib.optionals withQtGui [ qt5.qtbase ];
+
+  # For lib/cups/filter/kyofilter_pre_H.
+  # The source already contains a copy of pypdf3, but we use the Nix package
+  propagatedBuildInputs = with python3Packages; [ reportlab pypdf3 setuptools ];
+
+  installPhase = let
+      ver_maj_min = builtins.concatStringsSep "." (lib.lists.take 2 (builtins.splitVersion version));
+  in ''
+    # unpack the debian archive
+    ar p ${src}/Debian/${region}/kyodialog_${platform}/kyodialog_${kyodialog_version}_${platform}.deb data.tar.gz | tar -xz
+
+
+    # allow cups to find the ppd files
+    mkdir -p $out/share/cups/model
+    mv ./usr/share/kyocera${ver_maj_min}/ppd${ver_maj_min} $out/share/cups/model/Kyocera
+
+    # remove absolute path prefixes to filters in ppd
+    find $out -name "*.ppd" -exec sed -E -i "s:/usr/lib/cups/filter/::g" {} \;
+
+
+    mkdir -p $out/lib/cups/
+    mv ./usr/lib/cups/filter/ $out/lib/cups/
+    # for lib/cups/filter/kyofilter_pre_H
+    wrapPythonProgramsIn $out/lib/cups/filter "$propagatedBuildInputs"
+
+
+    install -Dm444 usr/share/doc/kyodialog/copyright $out/share/doc/${pname}/copyright
+
+  '' + lib.optionalString withQtGui ''
+    install -D usr/bin/kyoPPDWrite_H $out/bin/kyoPPDWrite_H
+    install -D usr/bin/kyodialog${ver_maj_min} $out/bin/kyodialog${ver_maj_min}
+
+    install -Dm444 usr/share/kyocera${ver_maj_min}/appicon_H.png $out/share/${pname}/icons/appicon_H.png
+
+    install -Dm444 usr/share/applications/kyodialog${ver_maj_min}.desktop $out/share/applications/kyodialog${ver_maj_min}.desktop
+    substituteInPlace $out/share/applications/kyodialog${ver_maj_min}.desktop \
+      --replace Exec=\"/usr/bin/kyodialog${ver_maj_min}\" Exec=\"$out/bin/kyodialog${ver_maj_min}\" \
+      --replace Icon=/usr/share/kyocera/appicon_H.png Icon=$out/share/${pname}/icons/appicon_H.png
+  '';
+  # Not installed:
+  #
+  # usr/bin/knmd
+  # usr/share/doc/kyodialog/changelog.Debian.gz
+  # usr/share/doc/kyodialog/changelog.gz
+  # usr/share/kyocera9.2/Python/PyPDF3-1.0.1/*
+  # etc/xdg/autostart/knmd.desktop
+  # etc/dbus-1/system.d/com.kyocera.knm.conf
+
+  meta = with lib; {
+    description = "CUPS drivers for several Kyocera printers";
+    homepage = "https://www.kyoceradocumentsolutions.com";
```

Comment:
Do we have a subpage that is not as generic?

Reply:
I retrieved the download URL for `src` via the driver search in the [download center](https://www.kyoceradocumentsolutions.com/download/index_en.html) for a [specific printer model (TASKalfa 5053ci)](https://www.kyoceradocumentsolutions.com/download/model_en.html?r=116&s=2&m=82&p=17), but I couldn't find a generic page for the drivers. 

[This](https://aur.archlinux.org/packages/kyocera-cups) ArchLinux pkg uses the download center URL.

Should I use one of them for `meta.homepage`?